ledger/participant-integration-api/src/test/suite/scala/platform/store/backend/h2/H2StorageBackendSpec.scala

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.h2
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class H2StorageBackendSpec extends AnyWordSpec with Matchers {
+
+  "H2StorageBackend" should {
+    "extractUserPasswordAndRemoveFromUrl" should {
```

Comment:
I would add a test with upper/mixed case `USER`/`PASSWORD` keys. E.g. (from the H2 docs):
```
jdbc:h2:file:~/sample;USER=sa;PASSWORD=123
```